pandas/io/excel/_calaminereader.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+
+from datetime import (
+    date,
+    datetime,
+    time,
+)
+from tempfile import NamedTemporaryFile
+from typing import (
+    Union,
+    cast,
+)
+
+from pandas._typing import (
+    FilePath,
+    ReadBuffer,
+    Scalar,
+    StorageOptions,
+)
+from pandas.compat._optional import import_optional_dependency
+from pandas.util._decorators import doc
+
+import pandas as pd
+from pandas.core.shared_docs import _shared_docs
+
+from pandas.io.common import stringify_path
+from pandas.io.excel._base import (
+    BaseExcelReader,
+    inspect_excel_format,
+)
+
+ValueT = Union[int, float, str, bool, time, date, datetime]
+
+
+class CalamineExcelReader(BaseExcelReader):
+    _sheet_names: list[str] | None = None
+
+    @doc(storage_options=_shared_docs["storage_options"])
+    def __init__(
+        self,
+        filepath_or_buffer: FilePath | ReadBuffer[bytes],
+        storage_options: StorageOptions = None,
+    ) -> None:
+        """
+        Reader using calamine engine (xlsx/xls/xlsb/ods).
+
+        Parameters
+        ----------
+        filepath_or_buffer : str, path to be parsed or
+            an open readable stream.
+        {storage_options}
+        """
+        import_optional_dependency("python_calamine")
+        super().__init__(filepath_or_buffer, storage_options=storage_options)
+
+    @property
+    def _workbook_class(self):
```

Comment:
The protocol must not be explicitly stated in code, but whatever is returned here is supposed to represent the concept of a Workbook. Not very familiar with calamine but the name `CalamineReader` sounds more responsible for loading data than representing a workbook concept

Reply:
Fixed.